uefi-test-runner/src/boot/memory.rs

```diff
@@ -63,45 +63,47 @@ fn alloc_alignment() {
 fn memory_map(bt: &BootServices) {
     info!("Testing memory map functions");
 
-    // Get the memory descriptor size and an estimate of the memory map size
-    let sizes = bt.memory_map_size();
-
-    // 2 extra descriptors should be enough.
-    let buf_sz = sizes.map_size + 2 * sizes.entry_size;
-
-    // We will use vectors for convenience.
-    let mut buffer = vec![0_u8; buf_sz];
+    /*
+    // Useful to check of OOM happens.
+    let mut maps = Vec::new();
+    for _ in 0..100000_u64 {
+        maps.push(bt.memory_map(MemoryType::LOADER_DATA).unwrap())
+    }
+    let _a = maps;*/
 
-    let mut memory_map = bt
-        .memory_map(&mut buffer)
-        .expect("Failed to retrieve UEFI memory map");
+    // Ensure that the memory of the UEFI memory map is always freed again.
```

Comment:
Can you add a bit more explanation here, how does this test that the memory map is freed?